packages/kbn-config-schema/src/types/type.ts

```diff
@@ -141,3 +150,17 @@ export abstract class Type<V> {
     return new SchemaTypeError(message || code, convertedPath);
   }
 }
+
+function recursiveGetSchemaStructure(internalSchema: AnySchema, path: string[] = []) {
+  const array: SchemaStructureEntry[] = [];
+  // Note: we are relying on Joi internals to obtain the schema structure (recursive keys).
+  // This is not ideal, but it works for now and we only need it for some integration test assertions.
+  // If it breaks in the future, we'll need to update our tests.
+  for (const [key, val] of (internalSchema as any)._ids._byKey.entries()) {
```

Comment:
😬 this is not ideal, but I'm not sure if we have a good alternative. I'm open to any other suggestions.

Reply:
We're in javascript, we don't have reflection, so I don't think there's any other way than using JOI's internals here.